experimental/AlgebraicStatistics/src/GaussianGraphicalModels.jl

```diff
@@ -372,3 +375,105 @@ function Base.show(io::IO, R::GaussianRing)
 end
 
+random_variables(GR::GaussianRing) = collect(1:nrows(covariance_matrix(GR)))
+
+@doc raw"""
+    ci_statements(GR::GaussianRing)
+
+Return all the `CIStmt` objects which can be formed on the `random_variables(GR)`.
+
+```jldoctest
+julia> GR = gaussian_ring(3)
+Gaussian ring over Rational field in 6 variables
+s[1, 1], s[1, 2], s[1, 3], s[2, 2], s[2, 3], s[3, 3]
+
+julia> ci_statements(GR)
+6-element Vector{CIStmt}:
+ [1 _||_ 2 | {}]
+ [1 _||_ 2 | 3]
+ [1 _||_ 3 | {}]
+ [1 _||_ 3 | 2]
+ [2 _||_ 3 | {}]
+ [2 _||_ 3 | 1]
+```
+"""
+ci_statements(GR::GaussianRing) = ci_statements(random_variables(GR))
+
+@doc raw"""
+    ci_ideal(GR::GaussianRing, stmts::Vector{CIStmt})
+
+Return the ideal for the conditional independence statements given by `stmts`.
+
+## Examples
+
+```jldoctest
+julia> GR = gaussian_ring(3)
+Gaussian ring over Rational field in 6 variables
+s[1, 1], s[1, 2], s[1, 3], s[2, 2], s[2, 3], s[3, 3]
+
+julia> ci_ideal(GR, [CI"1,2|", CI"1,2|3"])
+Ideal generated by
+  s[1, 2]
+  s[1, 2]*s[3, 3] - s[1, 3]*s[2, 3]
+```
+"""
+function ci_ideal(GR::GaussianRing, stmts::Vector{CIStmt})
+  eqs = Any[]
+  A = covariance_matrix(GR)
+  for stmt in stmts
+    k = length(stmt.K)
+    B = A[vcat(stmt.I, stmt.K), vcat(stmt.J, stmt.K)]
+    append!(eqs, minors(B, k+1))
+  end
+  return ideal(GR.ring, eqs)
+end
+
+@doc raw"""
+    ci_polynomial(GR::GaussianRing, stmt::CIStmt)
+    ci_polynomial(A::Generic.MatSpaceElem, stmt::CIStmt)
+
+Return the subdeterminant of the matrix `A` which corresponds to the
+elementary CI statement `stmt`. If a `GaussianRing` is given, its
+`covariance_matrix` is used.
+
+## Examples
+
+```jldoctest
+julia> GR = gaussian_ring(3)
+Gaussian ring over Rational field in 6 variables
+s[1, 1], s[1, 2], s[1, 3], s[2, 2], s[2, 3], s[3, 3]
+
+julia> ci_polynomial(GR, [CI"1,2|3"])
+s[1, 2]*s[3, 3] - s[1, 3]*s[2, 3]
+```
+"""
+function ci_polynomial(A::Generic.MatSpaceElem, stmt::CIStmt)
+  @assert is_elementary(stmt)
+  det(A[vcat(stmt.I, stmt.K), vcat(stmt.J, stmt.K)])
+end
+
+ci_polynomial(GR::GaussianRing, stmt::CIStmt) = ci_polynomial(covariance_matrix(GR), stmt)
+
+@doc raw"""
+    ci_structure(GR::GaussianRing; strategy::Symbol)
+
+Return the set of elementary CI statements which are satisfied by every distribution in the
+given Gaussian model. For this method to work, the ring needs to have one of the following
+methods:
+
+1. `parametrization(GR)` returning a parametrized covariance matrix. The CI structure is determined by evaluating its subdeterminants.
+2. `vanishing_ideal(GR)` returning the model's vanishing ideal. In this case, the subdeterminant for each CI statement is checked for contained in the vanishing ideal.
+
+The optional argument `strategy` can be used to give preference to one of the above methods using the values `:parametrization` or `:ideal`, respectively.
+"""
+function ci_structure(GR::GaussianRing; strategy::Symbol)
```

Comment:
this looks ok for now, ideally we get rid of the hasmethod function, so maybe just leave a note there for now?